packages/ai-openai/src/node/openai-language-model.ts

```diff
@@ -253,4 +239,92 @@ export class OpenAiModel implements LanguageModel {
             return new OpenAI({ apiKey: apiKey ?? 'no-key', baseURL: this.url });
         }
     }
+
+    protected processMessages(messages: LanguageModelRequestMessage[]): ChatCompletionMessageParam[] {
+        return this.openAiModelUtils.processMessages(messages, this.developerMessageSettings, this.model);
+    }
+}
+
+/**
+ * Utility class for processing messages for the OpenAI language model.
+ *
+ * Adopters can rebind this class to implement custom message processing behavior.
+ */
+@injectable()
+export class OpenAiModelUtils {
+
+    protected processSystemMessages(
+        messages: LanguageModelRequestMessage[],
+        developerMessageSettings: DeveloperMessageSettings
+    ): LanguageModelRequestMessage[] {
+        if (messages.length > 0 && messages[0].actor === 'system') {
+            if (developerMessageSettings === 'skip') {
+                return messages.slice(1);
+            } else if (developerMessageSettings === 'mergeWithFirstUserMessage') {
+                const systemMsg = messages[0];
+                const updatedMessages = messages.slice();
+                const userIndex = updatedMessages.findIndex((m, index) => index > 0 && m.actor === 'user');
```

Comment:
This seems a bit dangerous in a multi-turn conversation: where there are messages with the following order: `user`, `system`, `assistant`, `user`, `assistant`. We may skip over to the next user request essentially.

I think it'd be safer to find the next user and the next assistant message and if the next assistant message is closer than the next user message (or we receive -1 for user), we insert a user message as we do below.

This way, we prevent merging the system message into the next user/assistant response pair.